src/routes/posts.js

```diff
@@ -18,15 +22,14 @@ const fetchPosts = async (req, res, next) => {
 
 		res.status(200).json({
 			status: 'success',
-			data: feed.map((article) => ({
-				id: article.id,
-				userId: article.userId,
-				title: article.title,
-				content: article.content,
-				image: article.image,
-				published: article.published,
-				createdOn: article.createdAt
-			}))
+			data: feed.map((post) => {
+				if(post.type === 'article')
```

Comment:
@beblicarl cool work but we'll try to do a bit of refactoring here. This works no doubt but this is just to worry you haha

the aim is to use an object mapping instead of the if statement. sort of like the error mapping we do for the status codes. it helps to not have a lot of conditional statements based on different types of post


1. create a global variable in the same file likely called `typeTransformMap` or something better.
2. It will be an object which takes the post types as the object keys. so `article` will be a key and `gif` will also be a key
3. the `article` key should map to `transformArticleResponse` as its value
4. the `gif` key should map to `transformGifResponse` as its value
5. now instead of the if statements you will just have to call something like `typeTransformMap[post.type](post)`

so if its an article, it goes into the `typeTransformMap` uses the `post.type` to pick the `article` transformer and calls the `transformArticleResponse`

Reply:
sure, i will try it

Reply:
done